api/src/main/java/org/openmrs/Concept.java

```diff
@@ -710,12 +710,16 @@ public ConceptName getName(Locale locale, boolean exact) {
 		
 		ConceptName exactName = getNameInLocale(locale);
 		
-		if (exactName != null || exact) {
+		if (exactName != null) {
 			return exactName;
 		}
 		
-		//just get any name
-		return getName();
+		if (exact == false) {

```

Comment:
By the way, this should be "if (!exact)" -- it's bad style test == true or == false this way